runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java

```diff
@@ -253,6 +253,8 @@ public void getState(
       GetJobStateResponse response = GetJobStateResponse.newBuilder().setState(state).build();
       responseObserver.onNext(response);
       responseObserver.onCompleted();
+    } catch (StatusException e) {
```

Comment:
```suggestion
    } catch (StatusRuntimeException | StatusException e) {
```

Here and everywhere else otherwise the catch Exception will also convert the StatusRuntimeException to an `INTERNAL` status error.

Reply:
I updated `StatusException` to `StatusRuntimeException | StatusException` everywhere, but I'm a bit unclear on what's actually throwing the `StatusRuntimeException`.  I searched the code for `StatusRuntimeException` and I did not see a clear path from any of those cases to here. 

For example, looking at this code I didn't see anything suspicious inside the try/catch block:

```java
  @Override
  public void getState(
      GetJobStateRequest request, StreamObserver<GetJobStateResponse> responseObserver) {
    LOG.trace("{} {}", GetJobStateRequest.class.getSimpleName(), request);
    String invocationId = request.getJobId();
    try {
      JobInvocation invocation = getInvocation(invocationId);
      JobState.Enum state = invocation.getState();
      GetJobStateResponse response = GetJobStateResponse.newBuilder().setState(state).build();
      responseObserver.onNext(response);
      responseObserver.onCompleted();
    } catch (StatusRuntimeException | StatusException e) {
      responseObserver.onError(e);
    } catch (Exception e) {
      String errMessage =
          String.format("Encountered Unexpected Exception for Invocation %s", invocationId);
      LOG.error(errMessage, e);
      responseObserver.onError(Status.INTERNAL.withCause(e).asException());
    }
  }
```

Reply:
Since they are runtime exceptions, nothing is required to declare that they are thrown in their method signatures and can propagate up the call stack from any arbitrary location until they hit an appropriately scoped catch statement.

Without the change to also catch the StatusRuntimeException as part of the StatusException, they would be caught as part of the Exception block and converted to INTERNAL errors.